pennylane/workflow/qnode.py

```diff
@@ -730,7 +739,7 @@ def circuit(x):
         updated_qn._transform_program = qml.transforms.core.TransformProgram(self.transform_program)
         return updated_qn
 
-    def update_shots(self, shots: Union[int, Shots]) -> "QNode":
+    def update_shots(self, shots: int | Shots) -> "QNode":
```

Comment:
```suggestion
    def update_shots(self, shots: int | Shots) -> QNode:
```
Since we already had `from __future__ import annotations` I suppose this should work?

Reply:
Worked locally! https://github.com/PennyLaneAI/pennylane/pull/7745/commits/b50870f8fa281230db0be90e942d2f04eab3ac27